### ISSUE TITLE
bug: a.Mod(b) returns a negative number

### ISSUE THREAD

**Issue by basnijholt** [shopspring/decimal#210]
```go
	a := decimal.RequireFromString("0.499999999999999999")
	b := decimal.RequireFromString("0.25")
	fmt.Println(a.Mod(b))
```

prints `-0.000000000000000001`.

It should never be possible for the number to become negative.

**Comment by mwoss** [comment]
Thanks for spotting that bug! I will take a look. :)

**Comment by mwoss** [comment]
Completly forgot about this issue. This bug has been fixed twice in #258 and #317 :D
